samples/snippets/src/main/java/pubsublite/ConsumerExample.java

```diff
@@ -81,18 +83,23 @@ public static void consumerExample(
             .setAutocommit(true)
             .build();
 
+    Set<ConsumerRecord<byte[], byte[]>> hashSet = new HashSet<>();
     try (Consumer<byte[], byte[]> consumer = settings.instantiate()) {
+      // The consumer can only subscribe to the topic that it is associated to.
+      // If this is the only subscriber for this subscription, it will take up
+      // to 90s for the subscriber to warm up.
       consumer.subscribe(Arrays.asList(topicPath.toString()));
-      // It takes up to 90s for the subscriber to warm up.
       while (true) {
         ConsumerRecords<byte[], byte[]> records = consumer.poll(Duration.ofMillis(100));
         for (ConsumerRecord<byte[], byte[]> record : records) {
           long offset = record.offset();
-          byte[] value = record.value();
-          System.out.printf("Received %s: %s%n", offset, new String(value, StandardCharsets.UTF_8));
+          String value = Base64.getEncoder().encodeToString(record.value());
+          hashSet.add(record);
+          System.out.printf("Received %s: %s%n", offset, value);
         }
         // Early exit. Remove entirely to keep the consumer alive indefinitely.
-        if (!records.isEmpty()) {
+        if (hashSet.size() == 10) {
```

Comment:
nit, >=, you can get publish duplicates or messages from old subscriptions.

Reply:
Fixed.